master/handler/handler.go

```diff
@@ -32,13 +34,28 @@ type Handler interface {
 	// also sends message to client about issue
 	Validate() error
 	// Handle : obtains the message to be sent as response
-	Handle(pushTo rmq.Queue)
+	Handle(pushTo rmq.Queue, PushMetricsTo rmq.Queue)
```

Comment:
Instead of passing it to `Handle` pass it to `SetUp` and use it later

Reply:
I thought of that but it might not work.. because the errors we are trying to catch happen in the interface's handle method... so i have to pass the metrics queue to that method...  An Error such as 
```suggestion
	if err != nil {
		fmt.Printf("Failed to download image: %v\n", err)
		pushMetricsTo.PublishBytes(metricBytes)
		return
	}
```